packages/x-date-pickers/src/ClockPicker/ClockPointer.tsx

```diff
@@ -43,59 +70,40 @@ const ClockPointerThumb = styled('div')<{
 
 /**
  * @ignore - internal component.
- * TODO: Remove class
  */
-export class ClockPointer extends React.Component<ClockPointerProps> {
-  static getDerivedStateFromProps = (
-    nextProps: ClockPointerProps,
-    state: ClockPointer['state'],
-  ) => {
-    if (nextProps.type !== state.previousType) {
-      return {
-        toAnimateTransform: true,
-        previousType: nextProps.type,
-      };
-    }
-
-    return {
-      toAnimateTransform: false,
-      previousType: nextProps.type,
-    };
-  };
-
-  state = {
-    toAnimateTransform: false,
-    previousType: undefined,
-  };
+export function ClockPointer(inProps: ClockPointerProps) {
+  const props = useThemeProps({ props: inProps, name: 'MuiClockPointer' });
+  const previousType = React.useRef<ClockPickerView | null>(props.type);
+  React.useEffect(() => {
+    previousType.current = props.type;
+  }, [props.type]);
 
-  render() {
-    const { className, hasSelected, isInner, type, value, ...other } = this.props;
+  const { className, hasSelected, isInner, type, value, ...other } = props;
```

Comment:
Should it be put just after the `useThemeProps` to replace `props.type` by `type`?